webapp/plugins/pluggable.jsx

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+// EXPERIMENTAL - SUBJECT TO CHANGE
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class Pluggable extends React.PureComponent {
+    static propTypes = {
+        children: PropTypes.element.isRequired
+    }
+
+    render() {
+        const child = this.props.children.type;
```

Comment:
Ah crap, I meant React, not RN. I'm not sure if they allow you looking at the fields of the children. As long as it works now though, it's probably easy enough to fix later